x-pack/plugins/enterprise_search/public/applications/shared/layout/classic_nav_helpers.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ChromeNavLink, EuiSideNavItemTypeEnhanced } from '@kbn/core-chrome-browser';
+
+import {
+  ClassicNavItem,
+  GenerateNavLinkFromDeepLinkParameters,
+  GenerateNavLinkParameters,
+} from '../types';
+
+import { generateNavLink } from './nav_link_helpers';
+
+export const generateSideNavItems = (
```

Comment:
This function takes the `ClassicNavItem[]` and return what we need to render the side nav.

It's not 100% ideal for what I want but `generateNavLink` is very specific to `enterprise_search` and is using kea. My idea is to implement a more generic `enterprise_search` agnostic version of this function in another PR for a new `search_navigation` plugin. and that is the version that other plugins will use when project navigation is set to "classic".